packages/engine/src/flags.cairo

```diff
@@ -74,38 +72,38 @@ impl ScriptFlagsIntoU32 of Into<ScriptFlags, u32> {
     }
 }
 
-fn flag_from_string(flag: felt252) -> u32 {
+fn flag_from_string(flag: ByteArray) -> u32 {
     // TODO: To map and remaining flags
-    if flag == 'P2SH' {
+    if flag == "P2SH" {
         return ScriptFlags::ScriptBip16.into();
-    } else if flag == 'STRICTENC' {
+    } else if flag == "STRICTENC" {
         return ScriptFlags::ScriptVerifyStrictEncoding.into();
-    } else if flag == 'MINIMALDATA' {
+    } else if flag == "MINIMALDATA" {
         return ScriptFlags::ScriptVerifyMinimalData.into();
-    } else if flag == 'DISCOURAGE_UPGRADABLE_NOPS' {
+    } else if flag == "DISCOURAGE_UPGRADABLE_NOPS" {
         return ScriptFlags::ScriptDiscourageUpgradableNops.into();
-    } else if flag == 'DERSIG' {
+    } else if flag == "DERSIG" {
         return ScriptFlags::ScriptVerifyDERSignatures.into();
-    } else if flag == 'WITNESS' {
+    } else if flag == "WITNESS" {
         return ScriptFlags::ScriptVerifyWitness.into();
-    } else if flag == 'LOW_S' {
+    } else if flag == "LOW_S" {
         return ScriptFlags::ScriptVerifyLowS.into();
-    } else if flag == 'NULLDUMMY' {
+    } else if flag == "NULLDUMMY" {
         // TODO: Double check this
         return ScriptFlags::ScriptStrictMultiSig.into();
-    } else if flag == 'NULLFAIL' {
+    } else if flag == "NULLFAIL" {
         return ScriptFlags::ScriptVerifyNullFail.into();
-    } else if flag == 'SIGPUSHONLY' {
+    } else if flag == "SIGPUSHONLY" {
         return ScriptFlags::ScriptVerifySigPushOnly.into();
-    } else if flag == 'CLEANSTACK' {
+    } else if flag == "CLEANSTACK" {
         return ScriptFlags::ScriptVerifyCleanStack.into();
-    } else if flag == 'DISCOURAGE_UPGRADABLE_WITNESS' {
+    } else if flag == "DISCOURAGE_UPGRADABLE_WITNESS_PROGRAM" {
         return ScriptFlags::ScriptVerifyDiscourageUpgradeableWitnessProgram.into();
-    } else if flag == 'WITNESS_PUBKEYTYPE' {
+    } else if flag == "WITNESS_PUBKEYTYPE" {
         return ScriptFlags::ScriptVerifyWitnessPubKeyType.into();
-    } else if flag == 'MINIMALIF' {
+    } else if flag == "MINIMALIF" {
         return ScriptFlags::ScriptVerifyMinimalIf.into();
-    } else if flag == 'CHECKSEQUENCEVERIFY' {
+    } else if flag == "CHECKSEQUENCEVERIFY" {
         return ScriptFlags::ScriptVerifyCheckSequenceVerify.into();
     } else {
         return 0;
```

Comment:
please check if the conversion from `' '` to `" "` is neccessary

Reply:
Yes it is necessary, because cairo interprets  stuff inside `' '`  as numeric literals so I was getting error while converting them to ByteArray. It can only be converted to ByteArray when cairo interprets it as a string literal when it is inside `" "`.